cb-talent-development/employer-style-base
# EM-644 Fractal Pattern Library

Documentation is still in progress, but this is in a good place for review / feedback because:
* there are currently fewer files to review
* can give feedback that will impact how documentation is created

This story will not be ready for 'Done' with this PR. I'll continue documentation and adding dependencies in an additional branch.
 - committing the `build` introduced 1,000,000 files. I'll also commit that in a separate branch, but you can still view it on https://github.com/cb-talent-development/employer-style-base/tree/gh-pages
 - note button styles are not working on gh-pages though they work locally, will fix

**Purpose:**
> As a developer and/or designer I need the pattern library to be up to date so that it can be used for efficient design and development


* Adds [fractal](http://fractal.build) pattern library generator
* Adds [`node-sass-chokidar`](https://www.npmjs.com/package/node-sass-chokidar) sass compiler and watcher
  * outputs to `dist/css/employer_style_base.scss`
* So far, documents buttons and animation
* Updated README with Fractal notes. Please review and try it if you can. I won't know if these instructions are good enough until I can install on another computer or until others try them

**JIRA:**
https://cb-content-enablement.atlassian.net/browse/EM-644

**Changes:**
* Changes to setup
  * no change to how style-base is used in Employer
  * to use pattern library, you will need to see README notes

* Architectural changes
  * Each component now comes with its own directory and a set of documentation-related files. `_base.scss` is update accordingly

* Migrations
  *
  
* Library changes
  * will bump version

* Side effects
  * there should be none, but do check employer for weirdness

**Screenshots**
n/a

**QA Links:**
http://cb-talent-development.github.io/employer-style-base/

**How to Verify These Changes**
* Specific pages to visit
  * http://cb-talent-development.github.io/employer-style-base/

* Steps to take
  * Check it out!
  * More importantly check out the diffs of this PR, see if you understand what's going on, how documentation is added, how file structures have changed, and give feedback
  * Pull down this branch. Can you get Fractal running?
  * Check Employer locally. Have buttons or animations changed? (they shouldn't)
  

* Responsive considerations
  * n/a


**Relevant PRs/Dependencies:**


**Additional Information**
